### PR TITLE
Reduce timeline SQL requests

### PR DESCRIPTION
**Pull request by AdrienClairembault**
Reduce request on the ITIL timeline by avoiding "loading loops".

Before the changes, we had this king of loading loop:
- Load all followups for a ticket
- For each followup, instantiate a followup item
- `IITLFollowup::post_getFromDB()` is triggered,  which reload the parent ticket
- `Ticket::post_getFromDB()` is reload, which will loads its actors

On a test ticket with around 20 timeline event, this changes saved about ~300 SQL requests (1700 to 1400).

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !29020

### CODE REVIEW

File: src/CommonITILObject.php [hunk 1]
```diff
@@ -6888,70 +6920,80 @@ public function getTimelineItems(array $options = [])
...
+                    ($validation_row['status'] == CommonITILValidation::WAITING  ? "validation-waiting"  : "") .
+                    ($validation_row['status'] == CommonITILValidation::ACCEPTED ? "validation-accepted" : "") .
+                    ($validation_row['status'] == CommonITILValidation::REFUSED  ? "validation-refused"  : ""),
                     'item_action' => 'validation-request',
+                    'object'      => $validation,
                 ];
 
-                if (!empty($validation['validation_date'])) {
+                if (!empty($validation_row['validation_date'])) {
                     $timeline[$valitation_obj::getType() . "_" . $validations_id] = [
```
Comment (cedric-anne) [inline src/CommonITILObject.php:6975]: You could also pass `object` here.

File: src/ITILFollowup.php [hunk 2]
```diff
@@ -1016,4 +1023,26 @@ public function isFromSupportAgent()
...
+     * before loading the item, thus avoiding one useless DB query (or many more queries
+     * when looping on children items)
+     *
+     * TODO 10.1 move method and `item` property into parent class with
+     *
+     * @param $parent Parent item
+     *
+     * @return void
+     */
+    public function setParentItem($parent): void
```
Comment (cedric-anne) [inline src/ITILFollowup.php:1040]:
```suggestion
    final public function setParentItem(CommonITILObject $parent): void
```

File: src/ITILSolution.php [hunk 3]
```diff
@@ -416,4 +424,26 @@ public static function getIcon()
...
+     * before loading the item, thus avoiding one useless DB query (or many more queries
+     * when looping on children items)
+     *
+     * TODO 10.1 move method and `item` property into parent class with
+     *
+     * @param $parent Parent item
+     *
+     * @return void
+     */
+    public function setParentItem($parent): void
```
Comment (cedric-anne) [inline src/ITILSolution.php:441]:
```suggestion
    final public function setParentItem(CommonITILObject $parent): void
```

**Review by cedric-anne — APPROVED** [review]

**Comment by trasher** [comment]
It may seem a performance only fix, but it may bypass some logic also (getFromDB can do much more than just its name suggests; and getFromResultSet won't).
I have absolutely no  idea what can be impacted, and therefore if it can has an impact on timeline.

**Comment by AdrienClairembault** [comment]
> It may seem a performance only fix, but it may bypass some logic also (getFromDB can do much more than just its name suggests; and getFromResultSet won't). I have absolutely no idea what can be impacted, and therefore if it can has an impact on timeline.

We identified with @cedric-anne that `getFromResultSet ` should at the very least call the `post_getFromDB` method and that he would do the change in another PR.

**Comment by trasher** [comment]
> What I say is currently, we rely on `getFromDB` that does much more than just loading parent ticket. Switching to `getFromRestulSet` that does nothing may introduce issues.

Also, it my have no impact at all :D

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**